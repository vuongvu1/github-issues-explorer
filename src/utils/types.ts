export type IssueType = Array<{
  id: string;
  title: string;
  number: number;
  createdAt: string;
  author: {
    login: string;
  };
  comments: {
    totalCount: number;
  };
}>;
