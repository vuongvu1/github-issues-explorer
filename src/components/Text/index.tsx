import React from "react";
import SC from "./styles";

export enum TextType {
  H1 = "h1",
  H2 = "h2",
  BODY = "p",
}

interface Props {
  type?: TextType;
  color?: string;
}

const Text: React.FC<Props> = ({ type, color, children }) => {
  return (
    <SC.Container as={type || "div"} type={type} color={color}>
      {children}
    </SC.Container>
  );
};

export default Text;
