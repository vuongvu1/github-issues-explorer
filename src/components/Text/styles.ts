import styled, { css } from "styled-components";
import { TextType } from "./index";

const SC = {
  Container: styled.div<{ type?: TextType; color?: string }>(
    ({ theme, type, color }) => css`
      color: ${color || theme.palette.black};
      font-weight: normal;
      font-size: ${theme.fontSize.md};
      margin: 0;
      padding: 0;

      ${type === TextType.H1 &&
      css`
        font-weight: bold;
        font-size: ${theme.fontSize.lg};
      `}

      ${type === TextType.H2 &&
      css`
        font-weight: bold;
        font-size: ${theme.fontSize.md};
      `}

      ${type === TextType.BODY &&
      css`
        margin: 0;
        font-weight: normal;
        font-size: ${theme.fontSize.sm};
      `}
    `
  ),
};

export default SC;
