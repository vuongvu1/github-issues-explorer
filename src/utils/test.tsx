import { ReactNode } from "react";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import { MockedProvider } from "@apollo/react-testing";
import { BrowserRouter } from "react-router-dom";
import { theme } from "src/globalStyles";
import {
  IssueState,
  IssueOrderField,
  OrderDirection,
} from "src/generated/graphql";

const mockStore = configureMockStore();
const store = mockStore({
  repoSlice: {
    url: "https://github.com/reactjs/reactjs.org",
    currentPage: 1,
    searchParams: {
      name: "reactjs.org",
      owner: "reactjs",
      status: IssueState.Open,
      orderBy: IssueOrderField.CreatedAt,
      orderDir: OrderDirection.Desc,
      before: undefined,
      after: undefined,
      first: 10,
      last: undefined,
    },
  },
});

export const customRender = (Component: ReactNode) =>
  render(
    <ThemeProvider theme={theme}>
      <MockedProvider mocks={[]} addTypename={false}>
        <BrowserRouter>
          <Provider store={store}>{Component}</Provider>
        </BrowserRouter>
      </MockedProvider>
    </ThemeProvider>
  );
