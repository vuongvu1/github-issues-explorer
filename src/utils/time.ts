const ONE_YEAR_IN_SECONDS = 31536000;
const ONE_MONTH_IN_SECONDS = 2592000;
const ONE_DAY_IN_SECONDS = 86400;
const ONE_HOUR_IN_SECONDS = 3600;
const ONE_MINUTE_IN_SECONDS = 60;

export const timeSince = (date: Date) => {
  const seconds = Math.floor((new Date().getTime() - date.getTime()) / 1000);

  let interval = seconds / ONE_YEAR_IN_SECONDS;

  if (interval > 1) {
    return `${Math.floor(interval)} years`;
  }

  interval = seconds / ONE_MONTH_IN_SECONDS;
  if (interval > 1) {
    return `${Math.floor(interval)} months`;
  }

  interval = seconds / ONE_DAY_IN_SECONDS;
  if (interval > 1) {
    return `${Math.floor(interval)} days`;
  }

  interval = seconds / ONE_HOUR_IN_SECONDS;
  if (interval > 1) {
    return `${Math.floor(interval)} hours`;
  }

  interval = seconds / ONE_MINUTE_IN_SECONDS;
  if (interval > 1) {
    return `${Math.floor(interval)} minutes`;
  }

  return `${Math.floor(seconds)} seconds`;
};
