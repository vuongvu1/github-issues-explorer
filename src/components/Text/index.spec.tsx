import { cleanup } from "@testing-library/react";
import { customRender } from "src/utils/testUtils";
import Text, { TextType } from "./index";

describe("Text", () => {
  afterEach(cleanup);

  test("should match snapshot", () => {
    const TextH1 = customRender(<Text type={TextType.H1} />);
    const TextH2 = customRender(<Text type={TextType.H2} />);
    const TextBody = customRender(<Text type={TextType.BODY} />);

    expect(TextH1.container).toMatchSnapshot();
    expect(TextH2.container).toMatchSnapshot();
    expect(TextBody.container).toMatchSnapshot();
  });
});
