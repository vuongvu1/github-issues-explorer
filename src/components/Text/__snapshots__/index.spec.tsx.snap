// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should match snapshot 1`] = `
<div>
  <h1
    class="sc-bdvvaa iiiBEs"
    type="h1"
  />
</div>
`;

exports[`Text should match snapshot 2`] = `
<div>
  <h2
    class="sc-bdvvaa dFHBig"
    type="h2"
  />
</div>
`;

exports[`Text should match snapshot 3`] = `
<div>
  <p
    class="sc-bdvvaa cxIyPR"
    type="p"
  />
</div>
`;
